docs/coordinates/performance.inc.rst

```diff
@@ -56,3 +56,57 @@ properties::
 
 ..
   EXAMPLE END
+
+Improving Performance for Arrays of ``obstime``
+===============================================
+
+The most expensive operations when transforming between observer-dependent coordinate
+frames (e.g. ``AltAz``) and sky-fixed frames (e.g. ``ICRS``) are the calculation
+of astrometric values like precession and nutation matrices.
+
+If |skycoord| instances with a large ``obstime`` array are transformed,
+these calculations can be speed up by factors up to 100 and still keeping micro-arcsecond resolution
+by utilizing interpolation instead of calculating astrometric values for each individual point.
+
+This can be achieved through the ``erfa_astrom`` state and the ``ErfaAstromInterpolator``
+class like this
+
+..
+  EXAMPLE START
+  Improving performance for obstime arrays
+
+To use interpolation for the astrometric values in coordinate transformation, use::
+
+   >>> from astropy.coordinates import SkyCoord, EarthLocation, AltAz
+   >>> from astropy.coordinates.erfa_astrom import erfa_astrom, ErfaAstromInterpolator
+   >>> from astropy.time import Time
+   >>> from time import perf_counter
+   >>> import numpy as np
+   >>> import astropy.units as u
+
+
+   >>> # array with 10000 obstimes
+   >>> obstime = Time.now() + np.linspace(0, 6, 10000) * u.hour
+   >>> location = location = EarthLocation(lon=-17.89 * u.deg, lat=28.76 * u.deg, height=2200 * u.m)
+   >>> frame = AltAz(obstime=obstime, location=location)  # doctest:+REMOTE_DATA
+   >>> crab = SkyCoord.from_name('Crab')  # doctest:+REMOTE_DATA
+
+   >>> # transform with default transformation and print duration
+   >>> t0 = perf_counter()
+   >>> crab_altaz = crab.transform_to(frame)  # doctest:+REMOTE_DATA +IGNORE_WARNINGS
+   >>> print(f'Transformation took {perf_counter() - t0:.2f} s')  # doctest:+ELLIPSIS
+   Transformation took ... s
+
+   >>> # transform with interpolating astrometric values
+   >>> t0 = perf_counter()
+   >>> with erfa_astrom.set(ErfaAstromInterpolator(300 * u.s)):  # doctest:+REMOTE_DATA
+   ...     crab_altaz_interpolated = crab.transform_to(frame)  # doctest:+REMOTE_DATA +IGNORE_WARNINGS
+   >>> print(f'Transformation took {perf_counter() - t0:.2f} s')  # doctest:+ELLIPSIS
+   Transformation took ... s
+
+   >>> err = crab_altaz.separation(crab_altaz_interpolated)  # doctest:+REMOTE_DATA
+   >>> print(f'Mean error of interpolation: {err.to(u.microarcsecond).mean():.4f}')  # doctest:+ELLIPSIS +REMOTE_DATA
+   Mean error of interpolation: 0.0063 uarcsec
+
```

Comment:
I feel like this would encourage bad practices, doesn't it?

It's a pretty big change to how the transformations work and should be chosen appropriately at the point where it is used, right?

Reply:
I don’t think so; I can imagine someone wanting to use interpolation inside a whole script or interpreter process. As long as they understand the consequences I don’t see why we should discourage that.

Reply:
ok, added an example for this